packages/app/src/messages/reducers/messages.reducer.ts

```diff
@@ -269,5 +271,31 @@ export const messagesReducer = createReducer<MessagesState>(
           messageIdsInThreadMap: {},
         }
       })
+
+      .addCase(
+        fulfilledAction(DataSyncEvent.UpdateAllIndexes),
+        (state, action: UpdateAllIndexesAction) => {
+          return {
+            ...state,
+            threadMap: action.payload.threads,
+            messageMap: action.payload.messages,
+            messageIdsInThreadMap: Object.keys(action.payload.messages)
+              .map((messageKey) => {
+                return action.payload.messages[messageKey]
+              })
+              .reduce((prev: MessageIdsInThreadMap, message) => {
+                const messageIds: string[] = prev[message.threadId] ?? []
+                prev[message.threadId] = messageIds.find(
+                  (id) => id === message.id
+                )
+                  ? messageIds
+                  : [...messageIds, message.id]
+
+                return prev
+              }, {}),
+            threadsTotalCount: Object.keys(action.payload.messages).length - 1,
```

Comment:
```suggestion
            threadsTotalCount: Object.keys(action.payload.threads).length,
            threadsState: ResultState.Loaded,
```

Reply:
@dkarski It's how it looks like without `-1`
<img width="806" alt="Screen Shot 2022-01-21 at 8 26 44 AM" src="https://user-images.githubusercontent.com/21984930/150484617-ea20e883-b3c8-4abd-8849-f95d393b8545.png">